lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -934,54 +641,48 @@ protected function connect()
     /**
      * @throws MigrationException
      */
-    private function ensureOrganizeMigrationsIsCompatibleWithFinder()
+    private function ensureOrganizeMigrationsIsCompatibleWithFinder() : void
     {
-        if ( ! ($this->migrationFinder instanceof MigrationDeepFinderInterface)) {
+        if (! ($this->migrationFinder instanceof MigrationDeepFinder)) {
             throw MigrationException::configurationIncompatibleWithFinder(
                 'organize-migrations',
                 $this->migrationFinder
             );
         }
     }
 
-    /**
-     * Check if we should execute a migration for a given direction and target
-     * migration version.
-     *
-     * @param string  $direction The direction we are migrating.
-     * @param Version $version   The Version instance to check.
-     * @param string  $to        The version we are migrating to.
-     * @param array   $migrated  Migrated versions array.
-     *
-     * @return boolean
-     */
-    private function shouldExecuteMigration($direction, Version $version, $to, $migrated)
-    {
+    /** @param string[] $migrated */
+    private function shouldExecuteMigration(
+        string $direction,
+        Version $version,
+        string $to,
+        array $migrated
+    ) : bool {
         if ($direction === Version::DIRECTION_DOWN) {
-            if ( ! in_array($version->getVersion(), $migrated, true)) {
+            if (! in_array($version->getVersion(), $migrated)) {
                 return false;
             }
 
-            return $version->getVersion() > $to;
+            return $version->getVersion() > (int) $to;
         }
 
         if ($direction === Version::DIRECTION_UP) {
-            if (in_array($version->getVersion(), $migrated, true)) {
+            if (in_array($version->getVersion(), $migrated)) {
                 return false;
             }
 
-            return $version->getVersion() <= $to;
+            return $version->getVersion() <= (int) $to;
         }
```

Comment:
This could lead to crash if neither of conditions above passes. Although it shouldn't happen, I would add assert/exception here at least.